src/NServiceBus.AzureStoragePersistence.SagaDeduplicator/Index/EqualityComparers.cs

```diff
@@ -0,0 +1,71 @@
+namespace NServiceBus.AzureStoragePersistence.SagaDeduplicator.Index
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.WindowsAzure.Storage.Table;
+
+    public static class EqualityComparers
+    {
+        /// <summary>
+        /// Gets an <see cref="IEqualityComparer{Object}"/> casting down the passed object to the proper <paramref name="type"/>.
+        /// </summary>
+        public static IEqualityComparer<object> GetValueComparer(EdmType type)
```

Comment:
I suspect that edm type equality may result into false inequalities in some data sets, azure storage does not require the same property in different records to be of the same datatype. It's not likely (and you don't have to change this code right now), but keep this in mind if someone reports weird results in their datasets.